src/controllers/import.js

```diff
@@ -125,6 +127,63 @@ function ImportController(context) {
     }
   }
 
+  /**
+   * Extract the domain from a URL.
+   * @param inputUrl
+   * @return domain
+   */
+  function getDomain(inputUrl) {
+    const parsedUrl = new URL(inputUrl);
+    const parsedDomain = psl.parse(parsedUrl.hostname);
+    return parsedDomain.domain; // Extracts the full domain (e.g., example.co.uk)
+  }
+
+  /**
+   * Check if the URLs in urlList belong to any of the base domains.
+   * @param urlList
+   * @param baseDomainList
+   * @return {true} if all URLs belong to an allowed base domain
+   * @throws {ErrorWithStatusCode} if any URL does not belong to an allowed base domain
+   */
+  function isUrlInBaseDomains(urlList, baseDomainList) {
+    return urlList.every((inputUrl) => {
+      const urlDomain = getDomain(inputUrl);
+      const belongsToBaseDomain = baseDomainList.some((baseDomain) => urlDomain === baseDomain);
+      if (!belongsToBaseDomain) {
+        throw new ErrorWithStatusCode(`Invalid request: URL: ${inputUrl} not allowed`, STATUS_BAD_REQUEST);
+      }
+      return true;
+    });
+  }
+
+  /**
+   * Check if the URLs in the request belong to the allowed domains.
+   * @param urls
+   * @param scopes
+   */
+  function validateUrlsByDomain(urls, scopes) {
+    // We do not need to check the domains for users with scope: write_all_domains
+    if (scopes.some((scope) => scope.name === SCOPE.WRITE_ALL_DOMAINS)) {
+      return;
+    }
+
+    const allowedDomains = scopes
+      .filter((scope) => {
+        if (scope.name === SCOPE.WRITE) {
+          if (!scope.domains || scope.domains.length === 0) {
+            throw new ErrorWithStatusCode('Missing domain information', STATUS_UNAUTHORIZED);
+          }
```

Comment:
This check should be done as part of validateAccessScopes.  If we have a scope of SCOPE.WRITE (and does not have _WRITE__ALL_DOMAINS) then verify that the domains are set.